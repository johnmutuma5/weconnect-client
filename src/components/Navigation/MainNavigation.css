.top-bar {
    display: flex;
    justify-content: space-around;
    -webkit-box-sizing: border-box;
            box-sizing: border-box;
    background-color: #00396D;
    width: 100%;
    max-height: 50px;
    position: fixed;
    top: 0;
    z-index: 99;
    margin-bottom: 10px;
    -webkit-box-shadow: 3px 1px 3px #00080C;
            box-shadow: 3px 1px 3px #00080C;
}
