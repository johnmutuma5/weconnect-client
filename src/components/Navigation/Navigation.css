.top-bar {
    display: flex;
    flex-flow: column;
    align-items: center;
    justify-content: space-around;
    -webkit-box-sizing: border-box;
            box-sizing: border-box;
    background-color: #00396D;
    width: 100%;
    max-height: 100px;
    position: fixed;
    top: 0;
    z-index: 99;
    margin-bottom: 10px;
    -webkit-box-shadow: 3px 0px 10px #4C4502;
            box-shadow: 3px 0px 10px #4C4502;
}
