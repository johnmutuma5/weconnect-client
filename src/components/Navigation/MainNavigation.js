import React from 'react';
import Aux from '../../hoc/Aux';
import Toolbar from './Toolbar/Toolbar';
import SideDrawer from './SideDrawer/SideDrawer';
import './MainNavigation.css';

const mainNavigation = props => (
    <Aux>
        <SideDrawer />
        <div className='top-bar'>
            <Toolbar />
        </div>
    </Aux>
);

export default mainNavigation;
