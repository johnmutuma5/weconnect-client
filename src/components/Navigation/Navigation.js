import React from 'react';
import PropTypes from 'prop-types';
import Toolbar from './Toolbar/Toolbar';
import SideDrawer from './SideDrawer/SideDrawer';
import Backdrop from '../UI/Backdrop/Backdrop';
import Aux from '../../hoc/Aux';

import { toggleSideDrawer } from '../../store/actions/actions';
import './Navigation.css';


const mainNavigation = (props, context) => {
    const store = context.store;

    return (
        <Aux>
            <SideDrawer />
            <Backdrop
                active={ context.layoutState.sideDrawerOpen }
                id={ 'SideDrawerBackDrop' }
                click={ () => store.dispatch(toggleSideDrawer()) }/>

            <div className='top-bar'>
                <Toolbar />
            </div>
        </Aux>
    )
};

mainNavigation.contextTypes = {
    layoutState: PropTypes.object.isRequired,
    store: PropTypes.object.isRequired
}


export default mainNavigation;
