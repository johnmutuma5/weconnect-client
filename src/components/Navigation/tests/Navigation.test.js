import Navigation from '../Navigation';

describe('<Navigation /> Component', () => {
    let context;

    beforeEach(() => {
        context = {
            layoutState: {
                store: {},
                sideDrawerOpen: false
            }
        };
    });


    it('renders correctly', () => {
        const wrapper = shallow(<Navigation />, { context });
        expect(wrapper).toMatchSnapshot();
    });


    it('dispatches toggleSideDrawer action when sideDrawer Backdrop is clicked', () => {
        const backDropClickSpy = sinon.spy();
        const backdropWrapper = shallow(<Navigation />, { context })
                                    .find('#SideDrawerBackDrop')
                                    .dive(); // re-shallow the backdrop to setProps

        backdropWrapper
            .setProps({click: backDropClickSpy})
            .simulate('click');
        expect(backDropClickSpy.calledOnce).toBe(true);
    })
})
