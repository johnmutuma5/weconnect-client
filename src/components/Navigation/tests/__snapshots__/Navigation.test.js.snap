// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> Component renders correctly 1`] = `
<aux>
  <sideDrawer />
  <backDrop
    active={false}
    click={[Function]}
    id="SideDrawerBackDrop"
  />
  <div
    className="top-bar"
  >
    <toolbar />
    <Component />
  </div>
</aux>
`;
